seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/source/ElasticsearchSourceReader.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.seatunnel.connectors.seatunnel.elasticsearch.dto.source.ScrollResult;
+import org.apache.seatunnel.connectors.seatunnel.elasticsearch.dto.source.SourceIndexInfo;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class ElasticsearchSourceReader implements SourceReader<SeaTunnelRow, ElasticsearchSourceSplit> {
+
+    protected static final Logger LOG = LoggerFactory.getLogger(ElasticsearchSourceReader.class);
+
+    SourceReader.Context context;
+
+    private Config pluginConfig;
+
+    private EsRestClient esRestClient;
+
+    Deque<ElasticsearchSourceSplit> splits = new LinkedList<>();
+    boolean noMoreSplit;
+
+    private final long pollNextWaitTime = 1000L;
+
+    public ElasticsearchSourceReader(SourceReader.Context context, Config pluginConfig) {
+        this.context = context;
+        this.pluginConfig = pluginConfig;
+    }
+
+    @Override
+    public void open() {
+        esRestClient = EsRestClient.createInstance(this.pluginConfig);
+    }
+
+    @Override
+    public void close() throws IOException {
+        esRestClient.close();
+    }
+
+    @Override
+    public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
+        ElasticsearchSourceSplit split = splits.poll();
+        if (null != split) {
+            SourceIndexInfo sourceIndexInfo = split.getSourceIndexInfo();
+
+            ScrollResult scrollResult = esRestClient.searchByScroll(sourceIndexInfo.getIndex(), sourceIndexInfo.getSource(), sourceIndexInfo.getScrollTime(), sourceIndexInfo.getScrollSize());
+            outputFromScrollResult(scrollResult, sourceIndexInfo.getSource(), output);
+            while (scrollResult.getDocs() != null && scrollResult.getDocs().size() > 0) {
+                scrollResult = esRestClient.searchWithScrollId(scrollResult.getScrollId(), sourceIndexInfo.getScrollTime());
+                outputFromScrollResult(scrollResult, sourceIndexInfo.getSource(), output);
+            }
+        } else if (noMoreSplit) {
+            // signal to the source that we have reached the end of the data.
+            LOG.info("Closed the bounded ELasticsearch source");
+            context.signalNoMoreElement();
+        } else {
+            Thread.sleep(pollNextWaitTime);
+        }
+    }
+
+    private void outputFromScrollResult(ScrollResult scrollResult, List<String> source, Collector<SeaTunnelRow> output) {
+        int sourceSize = source.size();
+        for (Map<String, Object> doc : scrollResult.getDocs()) {
+            SeaTunnelRow seaTunnelRow = new SeaTunnelRow(sourceSize);
+            for (int i = 0; i < sourceSize; i++) {
+                Object value = doc.get(source.get(i));
+                seaTunnelRow.setField(i, String.valueOf(value));
```

Comment:
`String.valueOf(value)`  Convert all field to string ?

you can mapping elasticsearch datatype to seatunnel datatype ?

reference
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/serialize

Reply:
ok